### DISCUSSION TITLE
Date Formatting

### DISCUSSION THREAD

**Discussion by 0204xls** [microsoft-search/pnp-modern-search#3112] [Q&A] [answered]
Quick question about date formatting: 

How can I get a RefinableDateXXYY property to display results as DD/MM/YYYY? Right now, they come out looking like `2023-06-29T22:00:00Z`. I tried using the following handlebar expression but it does nothing: `{{getDate RefinableDate00 "DD/MM/YYYY" 0 true}}`. 

Any help would be much appreciated!

**Comment by wobba** [comment]
You don't share your handlebar template, but likely you need `{{getDate item.RefinableDate00 'DD/MM/YYYY' 0 true}}`  or use slots.

**Reply by 0204xls** [reply]
Hi Mikael. Thanks for your response. I tried {{getDate item.RefinableDate00 'DD/MM/YYYY' 0 true}} as the column data and still shows the long-form date. RefinableDate00 also sometimes holds more than one value in case that affects anything. 

My results template is as follows: 

<content id="data-content">

    <style>
        /* Insert your CSS overrides here */
    </style>

    <div class="template">

        {{#if @root.properties.showSelectedFilters}}
            <pnp-selectedfilters 
                data-filters="{{JSONstringify filters.selectedFilters 2}}" 
                data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}" 
                data-instance-id="{{filters.instanceId}}"
                data-operator="{{filters.filterOperator}}"
                data-theme-variant="{{JSONstringify @root.theme}}"
            >
            </pnp-selectedfilters>
        {{/if}}
        
        {{#if @root.properties.showResultsCount}}
            <div class="template--resultCount">
                <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
            </div>
        {{/if}}

        <pnp-detailslist 
            data-items="{{JSONstringify data.items}}" 
            data-columns-configuration="{{JSONstringify properties.layoutProperties.detailsListColumns}}" 
            data-show-file-icon="{{properties.layoutProperties.showFileIcon}}" 
            data-file-extension-field="{{properties.layoutProperties.fieldIconExtension}}"  
            data-is-container-field="{{@root.slots.IsFolder}}"
            data-is-compact="{{properties.layoutProperties.isCompact}}"
            data-group-by="{{properties.layoutProperties.groupByField}}"
            data-groups-collapsed="{{properties.layoutProperties.groupsCollapsed}}"
            data-context="{{JSONstringify @root}}"
            data-instance-id="{{@root.instanceId}}"
            data-theme-variant="{{JSONstringify @root.theme}}"
            data-allow-item-selection="{{properties.itemSelectionProps.allowItemSelection}}"
            data-allow-multi={{properties.itemSelectionProps.allowMulti}}
            data-selected-keys="{{JSONstringify @root.selectedKeys}}"
            data-fields="{{JSONstringify @root.properties.dataSourceProperties.sortList}}" 
            data-default-selected-field="{{sort.selectedSortFieldName}}" 
            data-default-direction="{{sort.selectedSortDirection}}"
        >
        </pnp-detailslist>

        {{#if @root.properties.paging.showPaging}}

            {{#gt @root.data.totalItemsCount @root.properties.paging.itemsCountPerPage}}
            
                <pnp-pagination 
                    data-total-items="{{@root.data.totalItemsCount}}" 
                    data-hide-first-last-pages="{{@root.properties.paging.hideFirstLastPages}}"
                    data-hide-disabled="{{@root.properties.paging.hideDisabled}}"
                    data-hide-navigation="{{@root.properties.paging.hideNavigation}}"
                    data-range="{{@root.properties.paging.pagingRange}}" 
                    data-items-count-per-page="{{@root.properties.paging.itemsCountPerPage}}" 
                    data-current-page-number="{{@root.paging.currentPageNumber}}"
                    data-theme-variant="{{JSONstringify @root.theme}}"
                >
                </pnp-pagination>
            {{/gt}}
            
        {{/if}}
    </div>
</content>

<content id="placeholder-content">
    <style>
        /* Insert your CSS overrides here */
    </style>

    <div class="placeholder">
        {{!-- #if showResultsCount --}}
            <div class="template--resultCount">
                <span class="placeholder--shimmer placeholder--line" style="width: 20%"></span>
            </div>
        {{!-- /if --}}
        <pnp-detailslist 
            data-columns-configuration="{{JSONstringify properties.layoutProperties.detailsListColumns}}" 
            data-show-file-icon="{{properties.layoutProperties.showFileIcon}}" 
            data-is-compact="{{properties.layoutProperties.isCompact}}" 
            data-show-shimmers="true"
            data-group-by="{{properties.layoutProperties.groupByField}}"
            data-groups-collapsed="{{properties.layoutProperties.groupsCollapsed}}"
            data-theme-variant="{{JSONstringify @root.theme}}"
            >
        </pnp-detailslist>
    </div>

</content>

**Reply by wobba** [reply]
Seems you have added `RefinableDate00` to the list of properties in `Selected properties` since you show data. I used this in the manage columns settings and works just fine.
<img width="523" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/51104/f106a438-bc3e-43ed-8739-0191c7daa7a3">

<img width="515" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/51104/92017334-8845-4d7f-b7ad-2e3e1d082c7d">

**Reply by 0204xls** [reply]
I ran a test with another date slot and it worked fine too. It could be the fact that there are multiple values in RefinableDate00 then. I will try to reduce crawled properties mapped to RD00 and see if it works then.

**Reply by wobba** [reply, marked as answer]
You might want to map to RefinableDateFirst00 instead which takes the value from the first cp with a value.

**Reply by 0204xls** [reply]
This did the trick. Great advice. 

It seems to be that the date reformatting does not work when there are multiple values returned in the refinable date property. Thank you again for the help!

**Reply by wobba** [reply]
Correct, as when you have a longer string with multiple dates it won't parse correctly.
